src/pal/inc/rt/rpc.h

```diff
@@ -19,8 +19,13 @@
 #define __RPC_USER
 #define __RPC_FAR
 
+#if defined(_MSC_VER) || defined(__llvm__)
 #define DECLSPEC_UUID(x) __declspec(uuid(x))
 #define MIDL_INTERFACE(x)   struct DECLSPEC_UUID(x) DECLSPEC_NOVTABLE
+#else
+#define DECLSPEC_UUID(x)
```

Comment:
Can we use this definition everywhere, or are there build breaks with it?

Reply:
yes, gcc doesn't like declspec. I'm planning to stub it out during compilation with preprocessor.

Reply:
> yes, gcc doesn't like declspec. I'm planning to stub it out during compilation with preprocessor.

Could you please submit the whole solution to the problem in a single PR?

Also, it would be highly preffered for GCC and LLVM to take the same path.

Reply:
Sure, let's use this as a start. Current version is no where close to production quality but could give you the whole picture.